### PR TITLE
Replace std::bitset with customize bitset for less memory usage

### LINKED ISSUE #2328: Replace std::bitset<16> in ColumnReader with uint16_t
I thought the size of `std::bitset<16>` is 2 bytes, but it turns out it's 8 bytes
```
#include <bitset>
#include <iostram>

int main() {
    std::cout << sizeof(std::bitset<2>) << std::endl;
    std::cout << sizeof(std::bitset<8>) << std::endl;
    std::cout << sizeof(std::bitset<16>) << std::endl;
    std::cout << sizeof(std::bitset<32>) << std::endl;
    std::cout << sizeof(std::bitset<64>) << std::endl;
    std::cout << sizeof(std::bitset<128>) << std::endl;
    return 0;
}
```
Output:
```
8
8
8
8
8
16
```

So using `uint16_t` instead of `std::bitset<16>` can give us a smaller `ColumnReader`

https://github.com/StarRocks/starrocks/blob/a0343a78460c9f878ee424a857d44dc9f67a4543/be/src/storage/rowset/segment_v2/column_reader.h#L255

### PR DESCRIPTION
**Pull request by sduzh**
Fix issue #2328

### CODE REVIEW

**Comment by sduzh** [comment]
run starrocks_be_unittest

**Comment by chaoyli** [comment]
How much memory can this pull request can save?

File: be/src/storage/rowset/column_reader.h [hunk 1]
```diff
@@ -173,6 +179,23 @@ class ColumnReader {
         Reader* reader;
     };
 
+    template <int Size>
+    struct SmallBitset {
```
Comment (chaoyli) [inline be/src/storage/rowset/column_reader.h:183]: Could you move this util class to util/ directory?

**Review by chaoyli — CHANGES_REQUESTED** [review]

File: be/src/storage/rowset/column_reader.h (hunk 1 above)
Comment (imay) [reply be/src/storage/rowset/column_reader.h:183]: Give some comments for this class, for example to explain what does the Size stand for.

File: be/src/storage/rowset/column_reader.h [hunk 2]
```diff
@@ -101,6 +100,13 @@ class ColumnReader {
 
     ~ColumnReader();
 
+    // Disable copy and assignment
```
Comment (dirtysalt) [inline be/src/storage/rowset/column_reader.h:103]: I remember there is some macro in google library that can disable copy & assignment function.

**Pull request closed without merging**